### PR TITLE
Simplify nodejs client and wallet

### LINKED ISSUE #699: Investigate: Nodejs bindings wrapper type
Investigate if we can change the bindings wrappers into something more usable.
Currently its a type wrapped in a struct, that aint pretty

See #679

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

This PR cleans up the client and wallet nodejs rust bindings.

## Links to any relevant issues

Closes #699

Replaces https://github.com/iotaledger/iota-sdk/pull/1056

### CODE REVIEW

**Review by kwek20 — DISMISSED** [review]

**Review by kwek20 — DISMISSED** [review]

File: bindings/wasm/lib/bindings.ts [hunk 1]
```diff
@@ -20,6 +21,15 @@ const callUtilsMethod = (method: __UtilsMethods__): any => {
     }
 };
 
+const getSecretManagerFromWallet = (
+    method: WalletMethodHandler,
+): SecretManagerMethodHandler => {
+    // TODO figure out why this one is extensible but client isnt
```
Comment (Thoralf-M) [inline bindings/wasm/lib/bindings.ts:27]: Is client from wasm really not extensible or is it only in the nodejs binding?

Comment (kwek20) [reply bindings/wasm/lib/bindings.ts:27]: tbh i am not sure why, i jsut know that this fixes the issue. maybe it has always existed even. Ive already spend soo much time on it though, not sure its worth it to do more

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

**Review by kwek20 — DISMISSED** [review]

**Review by DaughterOfMars — CHANGES_REQUESTED** [review]
Changes are breaking. Can't merge at 1.1

**Comment by kwek20** [comment]
> Changes are breaking. Can't merge at 1.1

If you just mean the Error("") vs "", i can change that and add a todo? Sync methods can also be awaited, so user code doesnt change

**Comment by thibault-martinez** [comment]
> > Changes are breaking. Can't merge at 1.1
> 
> If you just mean the Error("") vs "", i can change that and add a todo? Sync methods can also be awaited, so user code doesnt change

@kwek20 @DaughterOfMars any update on this?

**Comment by DaughterOfMars** [comment]
I don't know if that statement is true but even if it is I don't really like the changes that were made

**Comment by kwek20** [comment]
The statement is true :) lets merge. at the very least its cleaner than current code. can always opt not to close the issue.

**Comment by Alex6323** [comment]
What's the status of this PR btw?

**Comment by DaughterOfMars** [comment]
Waiting for @kwek20 to rebase it on 2.0

**Comment by kwek20** [comment]
closed in favour of the 2.0 PR

**Pull request closed without merging**